src/CommunityToolkit.Maui.Core/Views/DrawingView/Service/DrawingViewService.android.cs

```diff
@@ -19,13 +19,14 @@ public static class DrawingViewService
 	/// <param name="lines">Drawing lines</param>
 	/// <param name="imageSize">Maximum image size. The image will be resized proportionally.</param>
 	/// <param name="background">Image background</param>
+	/// <param name="token"><see cref="CancellationToken"/></param>
 	/// <returns>Image stream</returns>
-	public static ValueTask<Stream> GetImageStream(in IList<IDrawingLine> lines,
-												   in Size imageSize,
-												   in Paint? background)
+	public static ValueTask<Stream> GetImageStream(IList<IDrawingLine> lines, Size imageSize, Paint? background, CancellationToken token = default)
```

Comment:
Maybe we should just return stream instead of ValueTask.

Reply:
So any of the underlying methods require async/await? I don’t have my computer in front of me right now, but I think the Tizen implementation uses async/await.

Reply:
Yup - confirmed that `DrawingVewService.tizen.cs` uses async/await:

https://github.com/CommunityToolkit/Maui/blob/ece95a00bc2e959fbfd952877be32890800277fd/src/CommunityToolkit.Maui.Core/Views/DrawingView/Service/DrawingViewService.tizen.cs#L21-L46

Albiet, it is a self-induced; i.e. we are purposefully pushing the logic to a background thread. 

I hesitate to remove the async/await logic until we loop in Jay to do some performance testing. Perhaps we could do that in a future Proposal / PR? I'd really like to get this PR merged before our .NET 8 release since contains many breaking changes to our API surface.

Reply:
<img width="572" alt="image" src="https://github.com/CommunityToolkit/Maui/assets/13558917/ad845b57-3c25-4a90-bbc5-846435710df1">

I [followed the link in the code comment](https://github.com/CommunityToolkit/Maui/pull/692#pullrequestreview-1150202727), and Seungkeun Lee (software engineer at Tizen) recommended deferring to a thread-pool for the Tizen implementation. 

https://github.com/CommunityToolkit/Maui/pull/692/files/c431204d443256918516d508905883cd60c2f146..947056d463d90f2620ece5ed7fe79cf6531f9674#diff-a26ec24614bb6020677735a034f76ab28d2279a4afd7678f7c22357888ce3c92